### DISCUSSION TITLE
ナビゲーション階層について

### DISCUSSION THREAD

**Discussion by chippy-ao** [tatsutakein-jp/asis-app#313] [General]
`parentNavigatorKey` の指定があえてされていそうな部分がいくつかありました。

https://github.com/tatsutakein-jp/asis-app/blob/63a7653e7fce6bcb54628efcbd118a9a086eac25/app/lib/router/routes/quest_list_route.dart#L32-L47

各タブにおけるヒエラルキートップの `NavigatorState` 以外のものを指定していることで、遷移した際にボトムタブを隠す形で遷移先ページが表示されます。

Material Design のガイドに準じようとは思いますが、個人的には [WWDC](https://developer.apple.com/videos/play/wwdc2022/10001) にあるように、モーダル等の排他処理が好ましいもの以外で他機能を制限させるのはあまりよくないと思っています。

**Comment by tatsutakein** [comment]
ボトムタブがあると関心事が薄れて微妙かなと思っての実装でしたが，強いこだわりがあるわけではないのでボトムタブが見える形での実装に戻すでよいと思います．

**Reply by chippy-ao** [reply]
なるほど、盲点 :flushed:
逆に気持ちが揺らいできた。
